pkg/apis/pubsub/v1alpha1/pull_subscription_validation.go

```diff
@@ -49,6 +55,14 @@ func (current *PullSubscriptionSpec) Validate(ctx context.Context) *apis.FieldEr
 			errs = errs.Also(err.ViaField("transformer"))
 		}
 	}
+
+	if current.RetentionDuration != nil {
+		// If set, RetentionDuration Cannot be longer than 7 days or shorter than 10 minutes.
+		if *current.RetentionDuration < minRetentionDuration || *current.RetentionDuration > maxRetentionDuration {
+			errs = errs.Also(apis.ErrInvalidValue(current.RetentionDuration, "retentionDuration"))
```

Comment:
We have an OOB error function you should use for bounded values.

Reply:
Found it, used it.